Include/pyhash.h

```diff
@@ -16,6 +16,27 @@ typedef struct {
 PyAPI_FUNC(PyHash_FuncDef*) PyHash_GetFuncDef(void);
 #endif
 
+/* Prime multiplier used in string and various other hashes. */
+#define _PyHASH_MULTIPLIER 1000003UL  /* 0xf4243 */
+
+/* Parameters used for the numeric hash implementation.  See notes for
+   _Py_HashDouble in Python/pyhash.c.  Numeric hashes are based on
+   reduction modulo the prime 2**_PyHASH_BITS - 1. */
+
+#if SIZEOF_VOID_P >= 8
+#  define _PyHASH_BITS 61
+#else
+#  define _PyHASH_BITS 31
+#endif
+
+#define PyUnstable_PyHASH_BITS _PyHASH_BITS
+
+#define _PyHASH_MODULUS (((size_t)1 << _PyHASH_BITS) - 1)
+#define PyUnstable_PyHASH_MODULUS _PyHASH_MODULUS
+#define _PyHASH_INF 314159
+#define PyUnstable_PyHASH_INF _PyHASH_INF
+#define _PyHASH_IMAG _PyHASH_MULTIPLIER
+#define PyUnstable_PyHASH_IMAG _PyHASH_IMAG
```

Comment:
> Your change adds these API to the limited C API. I don't think that we want that.

In past hashing of numeric types was changed in minor releases (3.x.0).  Are there reasons we don't want this freedom in future?

> I suggest waiting until this PR is merged before considering exposing these constants to the public C API.

Sound reasonable.

Reply:
> In past hashing of numeric types was changed in minor releases (3.x.0). Are there reasons we don't want this freedom in future?

It was proposed to create a C API Working Group. The Steering Council [has to take a decision](https://github.com/python/steering-council/issues/210). For now, no new API should be added to the limited C API.

See https://github.com/capi-workgroup/api-evolution/issues/41 and https://github.com/capi-workgroup/api-evolution/issues/42

Reply:
Ok, added as public.  Docs moved to the hash.rst, news entry also added.  I don't think this could introduce merge conflicts with https://github.com/python/cpython/pull/112095